server/chat-commands/info.ts

```diff
@@ -13,14 +13,17 @@ import {YoutubeInterface} from '../chat-plugins/youtube';
 import {Utils} from '../../lib/utils';
 import {Net} from '../../lib/net';
 
-export function getCommonBattles(userID1: ID, user1: User | null, userID2: ID, user2: User | null) {
+export function getCommonBattles(
+	userID1: ID, user1: User | null, userID2: ID, user2: User | null, connection: Connection
+) {
 	const battles = [];
 	for (const curRoom of Rooms.rooms.values()) {
 		if (!curRoom.battle) continue;
 		if (
 			(user1?.inRooms.has(curRoom.roomid) || curRoom.auth.get(userID1) === Users.PLAYER_SYMBOL) &&
 			(user2?.inRooms.has(curRoom.roomid) || curRoom.auth.get(userID2) === Users.PLAYER_SYMBOL)
 		) {
+			if (connection) void curRoom.uploadReplay(connection.user, connection, "silent");
```

Comment:
Is there a `forPunishment` flag which should be used here? iirc there's some special rules for uploading replays for punishment purposes

Reply:
Yeah, it's `this.parse('/savereplay forpunishment')`